tests/chainer_tests/functions_tests/normalization_tests/test_batch_renormalization.py

```diff
@@ -1,218 +1,268 @@
+            decay = 0.9  # defaut value of F.batch_renormalization
+        else:
+            decay = None
+
+        x_arr, gamma_arr, beta_arr, gy_arr = inputs
+        x = chainer.Variable(x_arr)
+        gamma = chainer.Variable(gamma_arr[expander])
+        beta = chainer.Variable(beta_arr[expander])
+
+        x_mean = F.mean(x, axis=axis, keepdims=True)
+        x_var = F.mean((x - x_mean) ** 2, axis=axis, keepdims=True)
+        running_mean = self.running_mean.copy()
+        running_var = self.running_var.copy()
+        y = _naive_batch_renormalization(
+            x, gamma, beta,
+            x_mean,
+            x_var,
+            running_mean[expander],
+            running_var[expander],
+            self.rmax, self.dmax, self.eps,
+            decay)
+
+        # backward gradients
+        gx, ggamma, gbeta = self._compute_backward(
+            x, gamma, beta, y, chainer.Variable(gy_arr))
+        ggamma = numpy.squeeze(ggamma, axis)
+        gbeta = numpy.squeeze(gbeta, axis)
+
+        return (
+            y.array,
+            running_mean, running_var,
+            gx, ggamma, gbeta)
 
 
 @testing.parameterize(*testing.product({
     'ndim': [0, 1, 2, 3],
     'eps': [2e-5, 1e-1],
     'dtype': [numpy.float16, numpy.float32, numpy.float64],
 }))
-class TestFixedBatchRenormalization(unittest.TestCase):
+@inject_backend_tests_batch_renormalization()
+class TestFixedBatchRenormalization(testing.FunctionTestCase):
 
     def setUp(self):
-        self.gamma = numpy.random.uniform(.5, 1, (3,)).astype(self.dtype)
-        self.beta = numpy.random.uniform(-1, 1, (3,)).astype(self.dtype)
-        self.expander = (None, Ellipsis) + (None,) * self.ndim
-
-        self.rmax = self.dtype(3)
-        self.dmax = self.dtype(5)
-
-        shape = (5, 3) + (2,) * self.ndim
-        self.x = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
-        self.gy = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
-        self.aggr_axes = (0,) + tuple(six.moves.range(2, self.ndim + 2))
-
-        self.mean = numpy.random.uniform(-1, 1, (3,)).astype(self.dtype)
-        self.var = numpy.random.uniform(
-            0.5, 1, (3,)).astype(self.dtype)
-        self.args = [self.x, self.gamma, self.beta, self.mean, self.var]
-        self.train = False
-        self.check_forward_options = {'atol': 1e-4, 'rtol': 1e-3}
-        self.check_backward_options = {'dtype': numpy.float64}
         if self.dtype == numpy.float16:
-            self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-2}
-            self.check_backward_options = {
-                'dtype': numpy.float64, 'atol': 1e-3, 'rtol': 1e-2}
-
-    def _forward(self, *args):
+            self.check_forward_options.update({'atol': 5e-3, 'rtol': 1e-2})
+            self.check_backward_options.update({'atol': 5e-3, 'rtol': 1e-2})
+            self.check_double_backward_options.update(
+                {'atol': 5e-3, 'rtol': 1e-2})
+
+    def generate_inputs(self):
+        d = 3
```

Comment:
A detail but how about using a different variable name such as `channels` since `d` is used in the batch renormalization algorithm for a different meaning.